### PR TITLE
Feature en 7029 push preapproval outputs to salesforce

### PR DESCRIPTION
**Pull request by Mandon24**
## [type(scope):EN-7029](https://caminofinancial.atlassian.net/browse/EN-7029)

Creates AWS resources to send snowflake data in the form of events from eventbridge to salesforce

### TODO:
If you have some TODOs, please label this PR as 'in progress' and add them
here for others to follow:
- add tests to test against added sf code

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -7,7 +7,9 @@ frameworkVersion: '3'
 
 provider:
   name: aws
-  runtime: python3.10
+  runtime: python3.9
```
Comment (jjmonsalveg) [inline serverless.yml:10]: why are we coming back to a prev version?

File: src/handler.py [hunk 2]
```diff
@@ -56,6 +63,43 @@ def process_latest_pre_approvals() -> WorkflowOutput:
         for output in get_latest_pre_approvals_from_snowflake(db, last_sf_batch):
             current_sf_batch.loanapp_ids.append(output.LOANAPP_ID)
             # TODO: Send to salesforce
+            logger.info(f"Preapproval output instance: {output}")
+            fico = output.FICO
+            if fico:
+                if fico > 999:
+                    fico = ""
```
Comment (jjmonsalveg) [inline src/handler.py:70]:
```suggestion
            MY_AWESOME_FICO_NAME_WITH_MEANING = 999 
```

File: src/handler.py [hunk 3]
```diff
@@ -56,6 +63,43 @@ def process_latest_pre_approvals() -> WorkflowOutput:
...
+            event_detail = {
+                'payload': {
+                    'loanapp_id': output.LOANAPP_ID,
+                    'created_at': output.CREATED_AT.isoformat(),
+                    'requested_loan_amount': output.REQUESTED_LOAN_AMT,
+                    'interest_rate': output.INT_RATE,
+                    'risk_rate': output.RISK_RATE,
+                    'tenor': output.TENOR,
+                    'fico': fico,
```
Comment (jjmonsalveg) [inline src/handler.py:79]:
```suggestion
                    'fico':  "" if output.FICO and output.FICO > MY_AWESOME_FICO_NAME_WITH_MEANING else output.FICO,
```

File: src/handler.py [hunk 4]
```diff
@@ -56,6 +63,43 @@ def process_latest_pre_approvals() -> WorkflowOutput:
...
+                    'Detail': json.dumps(event_detail),
+                    'DetailType': 'preapproval-output',
+                    'EventBusName': os.getenv('EVENT_BUS_NAME'),
+                    'Source': 'snowflake-ms',
+                    'Time': datetime.now(timezone.utc)
+                }
+            ]
+            logger.info(f"Event data to be sent to eventbridge: {event_data}")
+            put_event_resp = eventbridge_client.put_events(Entries=event_data)
+            logger.info(put_event_resp)
```
Comment (jjmonsalveg) [inline src/handler.py:102]: A very important part is missing in this PR and that is when the transmission to the event bridge fails, you must place it in the batch failed list to try it first in the next execution. This part is missing here. cc: @Jvillegasd

**Review by jjmonsalveg — COMMENTED** [review]
I wanted to block the PR but maybe it is convenient to decide if we want to merge everything and make the case of failures separately (but still this sprint with the same tickets)

File: serverless.yml (hunk 1 above)
Comment (Mandon24) [reply serverless.yml:10]: I was working with 3.9 and wasnt aware johnny worked with 3.10. Will look to update

**Review by jjmonsalveg — APPROVED** [review]

**Review by jjmonsalveg — APPROVED** [review]

**Review by jjmonsalveg — APPROVED** [review]

**Review by vabadio — APPROVED** [review]

File: tests/test_handler.py [hunk 5]
```diff
@@ -149,10 +158,13 @@ def test_workflow_without_data(
...
     sqlite_dal: DataAccessLayer,
-    mocker
+    mocker,
+    monkeypatch,
+    custom_event_bus
 ):
     mocker.patch("src.handler.dal", sqlite_dal)
     mocker.patch("src.handler.dynamodb_res", dynamodb_resource)
+    monkeypatch.setenv("EVENT_BUS_NAME", custom_event_bus)
     from src.handler import main
```
Comment (CFCarlo1911) [inline tests/test_handler.py:168]: why import "main" in every test?

File: src/handler.py [hunk 6]
```diff
@@ -12,8 +13,12 @@
     LastSalesforceBatchRegistry,
 )
 from src.models.pre_approval_output import PreApprovalOutput
+from datetime import datetime, timezone
```
Comment (CFCarlo1911) [inline src/handler.py:16]: this could be on top of built-in packages.

**Review by CFCarlo1911 — APPROVED** [review]

**Pull request merged**